packages/react-instantsearch-nextjs/src/InitializePromise.tsx

```diff
@@ -26,7 +38,7 @@ export function InitializePromise() {
 
   const injectInitialResults = () => {
     let inserted = false;
-    const results = getInitialResults(search.mainIndex);
+    const results = getInitialResults(search.mainIndex, requestParamsList);
```

Comment:
I guess I don't really understand this code right now, but how come this can't be read from waitForResults? does it need to be set before the results are known or something?

Reply:
It's similar to what's in server.ts for the pages router, just that because we have access to the search instance outside of both `waitForResults()` and `injectInitialResults()`, it allows me to put the code that extracts the queries params before and directly reference them instead of passing them as a return value of promises.

Reply:
my question was more about the potential desynchronising of the search client wrapper in the future if we do any changes to that. Not sure if it matters